### DISCUSSION TITLE
Support for monorepo's such as TurboRepo or NX

### DISCUSSION THREAD

**Discussion by jeroenkroese** [medusajs/medusa#4424] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

The issue has been described here as well: https://github.com/medusajs/medusa/issues/2638

Im using NX to manage a monorepo and I have set up the Medusa Server in apps/medusa-server. The repo has 1 package.json in the root directory and the node_modules directory is also located in the project root directory.  When I run the command `medusa develop` from the project root directory, the command fails with the following error:

> babel:
>   src does not exist
> Error: Command failed: "/Users/jeroenkroese/my-repo/node_modules/.bin/babel" src -d dist
>     at checkExecSyncError (node:child_process:885:11)
>     at execSync (node:child_process:957:15)

    
 When I run the command `cd apps/medusa-server && medusa develop` I get the following error:

> /bin/sh: /Users/jeroenkroese/my-repo/apps/medusa-server/node_modules/.bin/babel: No such file or directory
> Error: Command failed: "/Users/jeroenkroese/my-repo/apps/medusa-server/node_modules/.bin/babel" src -d dist
>     at checkExecSyncError (node:child_process:885:11)
>     at execSync (node:child_process:957:15)
>     at /Users/jeroenkroese/my-repo/node_modules/@medusajs/medusa/dist/commands/develop.js:108:42

**Describe the solution you'd like**
Either a flag to be able point the cli to the directory that contains the medusa-config.js so that I can run the command from the project root directory, or a flag which allows to set the node_modules directory

**Comment by Toti91** [comment]
Seconded, we are having the same problem with NX workspace.

Did you find any workaround?

**Comment by jeroenkroese** [comment]
How I have it set up right now is I have Medusa in `./apps/medusa-server` with its own `package.json` and node-modules. To install the dependencies, cd into the the `./apps/medusa-server` and run `yarn` or `npm install`. I have configured the `targets` in the `project.json` as follows: 
```    
"serve": {
      "executor": "nx:run-commands",
      "options": {
        "command": "cd apps/medusa-server && npx medusa develop"
      }
    }
```

This allows me to run the commands such as `nx serve medusa-server` from the repo root.
